qi-doc/scribblings/intro.scrbl

```diff
@@ -69,6 +69,29 @@ Qi is a hosted language on the @hyperlink["https://racket-lang.org/"]{Racket pla
+
+Furthermore, Qi packages on the Racket package index point to this main branch on the source host, so that running @racket[raco pkg install qi] or @racket[raco pkg update qi] will always get the version on the @racket[main] branch, reflecting the latest improvements.
+
+This doesn't mean that you must use this version, however. The expressiveness of modern version control systems allows us to define diverse versioning protocols directly on the versioning backend to best support diverse usage needs. Towards this end, Qi follows these conventions:
+
+@itemlist[#:style 'ordered
+  @item{Each significant new release has a tagged version, which is static and immutable.}
+  @item{Each legacy release has a "maintenance" branch that will be stable without any backwards incompatible changes or new features, and will be supported with bug fixes as needed.}
+]
+
+Together, these practices decouple development from use, effectively eliminating the problem of backwards compatibility. Specifically, Qi may occasionally change in a way that might traditionally be labeled "backwards-incompatible," but by relying on a version tag with either of the above semantics, such a change would not affect your application unless you are interested in the new features and until such a time as you are ready to upgrade. Thus, it gives users flexibility and stability without compromising the freedom to innovate and remedy past missteps for developers of Qi.
+
+In case you need to rely on a version with either of the above semantics, we recommend declaring a Git @tech{package source} in your @racket[info.rkt], instead of a Racket @tech{package source}. Like this:
+
+@codeblock{
+  (define deps '("git://github.com/drym-org/qi.git#v5.0"))
+}
+
+Now, traditionally, we may have grown accustomed to depending on a certain version of a package "or newer" (as are the semantics of using a Racket, rather than Git, @tech{package source}), so that we never have to update the dependency specification to get the latest improvements. We believe that this is a fragile convention that simultaneously overburdens development while threatening application stability. It's inadvisable for the same reason that mutability in programs is inadvisable, that is, introducing superfluous coupling and incurring the attendant risks. After all, any introduced bug is technically backwards-incompatible, as, indeed, is the fix for the bug! On the other hand, the branch strategy above supports these semantics to a sensible extent -- that of receiving necessary bug fixes, but not gratuitous "improvements" that may unwittingly break your application, even if they aren't intended to be backwards-incompatible.
```

Comment:
> After all, any introduced bug is technically backwards-incompatible, as, indeed, is the fix for the bug!

This is _at least_ somewhat controversial, and typically depends on the bug and its blast radius. I think different groups handle this differently, so it might be worth calling out that this is Qi's stance rather than a broad point.

In particular: at `$DAYJOB` we often rely on patch (or minor) versions to get bugfixes regularly in our dependencies.

Reply:
The former rather than the latter (I personally try to avoid pinning to `x.y.z`, because then every time a CVE is announced we have to go update those pins; instead, we just seamless update to the newest `z`, _possibly_ by re-locking dependencies in systems that do that).

Anyway, the text has moved and I didn't look to see if it still considers bug fixes backwards-incompatible changes.